### PR TITLE
[IMP] sale_management: default template by company.

### PR DESCRIPTION
**Pull request by Feyensv**
Make the default sales order template defined by company and not by instance anymore.

[Task ID - 2092848](https://www.odoo.com/web?#id=2092848&action=333&active_id=1725&model=project.task&view_type=form&menu_id=4720)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_management/models/res_config_settings.py [hunk 1]
```diff
@@ -8,11 +8,35 @@ class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
     group_sale_order_template = fields.Boolean("Quotation Templates", implied_group='sale_management.group_sale_order_template')
-    default_sale_order_template_id = fields.Many2one('sale.order.template', default_model='sale.order', string='Default Template')
+    company_so_template_id = fields.Many2one(
```
Comment (Feyensv) [inline addons/sale_management/models/res_config_settings.py]: Keep the field name.

File: addons/sale_management/models/res_config_settings.py [hunk 2]
```diff
@@ -8,11 +8,35 @@ class ResConfigSettings(models.TransientModel):
...
     module_sale_quotation_builder = fields.Boolean("Quotation Builder")
 
     @api.onchange('group_sale_order_template')
     def _onchange_group_sale_order_template(self):
         if not self.group_sale_order_template:
             self.module_sale_quotation_builder = False
-            self.default_sale_order_template_id = False
+
+    @api.model
+    def create(self, values):
```
Comment (dbo-odoo) [inline addons/sale_management/models/res_config_settings.py]: seems louche, should perhaps be a override of `default_get` and `execute` instead

File: addons/sale_management/models/sale_order.py [hunk 3]
```diff
@@ -10,9 +10,14 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    def _default_sale_order_template_id(self):
```
Comment (dbo-odoo) [inline addons/sale_management/models/sale_order.py]: a default that depends on other defaults (company), shouldn't it be a default_get override instead?

File: addons/sale_management/models/sale_order_template.py [hunk 4]
```diff
@@ -53,10 +53,15 @@ def create(self, vals_list):
 
     def write(self, vals):
         if 'active' in vals and not vals.get('active'):
-            template_id = self.env['ir.default'].get('sale.order', 'sale_order_template_id')
-            for template in self:
-                if template_id and template_id == template.id:
-                    raise UserError(_('Before archiving "%s" please select another default template in the settings.') % template.name)
+            companies = self.env['res.company'].sudo().search([('sale_order_template_id', 'in', self.ids)])
```
Comment (dbo-odoo) [inline addons/sale_management/models/sale_order_template.py]: i'd prefer to remove the default instead of blocking the operation

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**